### PR TITLE
feat!: Use CommandInput for commands

### PR DESCRIPTION
**Pull request by aawsome**
This re-allows to use a String for `password-command` or `warm-up-command`.

Breaking change: The former version to give the command in an array is no longer supported.
Instead of
```
password-command = ["echo", "password"]
```
please use either of
```
password-command = "echo password"
password-command = { command = "echo", args = ["password"] }
```

### CODE REVIEW

**Comment by simonsan** [comment]
Added to https://github.com/rustic-rs/docs/pull/88

File: crates/backend/src/rclone.rs [hunk 1]
```diff
@@ -150,12 +148,16 @@ impl RcloneBackend {
         let user = Alphanumeric.sample_string(&mut thread_rng(), 12);
         let password = Alphanumeric.sample_string(&mut thread_rng(), 12);
 
-        let mut rclone_command =
-            split(rclone_command.map_or("rclone serve restic --addr localhost:0", String::as_str))?;
-        rclone_command.push(url.as_ref().to_string());
+        let mut rclone_command = rclone_command.map_or(
+            "rclone serve restic --addr localhost:0".to_string(),
```
Comment (simonsan) [inline crates/backend/src/rclone.rs:152]: Would it make sense, to take this hardcoded command out as a constant `RCLONE_COMMAND` to `mod constants {}` in `rclone.rs` so we have this easily accessible and separated from the rest of the code?

File: crates/core/src/repository.rs [hunk 2]
```diff
@@ -356,11 +341,11 @@ impl<P> Repository<P, ()> {
         let mut be = backends.repository();
         let be_hot = backends.repo_hot();
 
-        if !opts.warm_up_command.is_empty() {
-            if opts.warm_up_command.iter().all(|c| !c.contains("%id")) {
+        if let Some(warm_up) = &opts.warm_up_command {
+            if !warm_up.args().iter().all(|c| !c.contains("%id")) {
```
Comment (simonsan) [inline crates/core/src/repository.rs:345]: Why is this negated now? 🤔

File: crates/backend/src/rclone.rs (hunk 1 above)
Comment (aawsome) [reply crates/backend/src/rclone.rs]: is changed.

File: crates/core/src/repository.rs (hunk 2 above)
Comment (aawsome) [reply crates/core/src/repository.rs]: Good catch - that was a typo.
I fixed it and also added an integration test which would have caught this.

**Review by simonsan — APPROVED** [review]

**Pull request merged**